apps/src/codebridge/Settings/SettingsDropdown.tsx

```diff
@@ -0,0 +1,140 @@
+
+import codebridgeI18n from '@cdo/apps/codebridge/locale';
+import {DEFAULT_FONT_SIZE_KEY, FontSize} from '@cdo/apps/lab2/constants';
+import useDropdownPosition from '@cdo/apps/lab2/hooks/useDropdownPosition';
+import {setEditorFontSize} from '@cdo/apps/lab2/redux/lab2ViewRedux';
+import {SignInState} from '@cdo/apps/templates/currentUserRedux';
+import useOutsideClick from '@cdo/apps/util/hooks/useOutsideClick';
+import {useAppSelector, useAppDispatch} from '@cdo/apps/util/reduxHooks';
+import {trySetSessionStorage} from '@cdo/apps/utils';
+import commonI18n from '@cdo/locale';
+
+import {useCodebridgeContext} from '../codebridgeContext';
+
+import moduleStyles from './settings-dropdown.module.scss';
+
+// fontSizeOptions contains a list of value/text from the FontSize enum,
+// e.g., [{value: 'Tiny', text: 'Tiny'}, {value: 'Small', text: 'Small'}, ...]
+const fontSizeOptions: SimpleDropdownProps['items'] = Object.keys(FontSize)
+  .filter(key => isNaN(Number(key))) // Filters out the reverse enum keys.
+  .map(key => ({
+    value: key,
+    text: key,
+  }));
+interface SettingsDropdownProps {
+  closeDropdown: () => void;
+  buttonRef: React.RefObject<HTMLDivElement>;
+}
+
+const SettingsDropdown: React.FunctionComponent<SettingsDropdownProps> = ({
+  closeDropdown,
+  buttonRef,
+}) => {
+  const dropdownRef = useOutsideClick<HTMLDivElement>(closeDropdown);
+  const editorFontSizeKey = useAppSelector(
+    state => state.lab2View.editorFontSizeKey
+  );
+  const selectedFontSizeKey = editorFontSizeKey || DEFAULT_FONT_SIZE_KEY;
+  const {signInState} = useAppSelector(state => state.currentUser);
+  const {levelProperties} = useCodebridgeContext();
+  const dispatch = useAppDispatch();
+  const [selectedValue, setSelectedValue] = useState(selectedFontSizeKey);
+
+  const getSelectedKey = (value: string) => value as keyof typeof FontSize;
+
+  const dropdownStyles = useDropdownPosition(buttonRef, dropdownRef);
+
+  const onTextEditorDropdownChange = async (value: string) => {
+    setSelectedValue(getSelectedKey(value));
+  };
+
+  const onSave = () => {
+    const selectedKey = getSelectedKey(selectedValue);
+    if (FontSize[selectedKey]) {
+      // We want the user preference for selected font size to persist across a session
+      // for signed-in users per app type.
+      if (signInState === SignInState.SignedIn) {
+        const sessionStorageKey = `${levelProperties.appName}CodeEditorFontSizeKey`;
+        trySetSessionStorage(sessionStorageKey, selectedKey);
+      }
+      dispatch(setEditorFontSize(selectedKey));
+    }
+    closeDropdown();
+  };
+
+  return createPortal(
+    <FocusTrap
+      focusTrapOptions={{
+        onDeactivate: closeDropdown,
+        clickOutsideDeactivates: true,
+      }}
+    >
+      <div
+        className={moduleStyles.settingsDropdown}
+        ref={dropdownRef}
+        role="dialog"
+        style={dropdownStyles}
+        aria-modal="true"
+        aria-label={commonI18n.settings()}
```

Comment:
can be a follow-up, but I think you can apply `data-theme="Dark"` and avoid some custom css/setting button color to white, similar to [this PR](https://github.com/code-dot-org/code-dot-org/pull/64972)

Reply:
Cool - will definitely do this as follow-up - thanks!

Reply:
I'll also address the other small comments in a fast follow-up (remove default constant, extra space). Thanks!